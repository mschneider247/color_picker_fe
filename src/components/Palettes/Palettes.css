#palettes {
  color: white;
}

#palettes_div {
  display: flex;
  flex-wrap: wrap;
  padding: 3%;
  width: 70%;
}

#add_new_palette {
  display: flex;
  width: 69%;
  padding: 1% 3% 3% 3%;
  background-color: rgba(0, 0, 0, 0.6);
  border-radius: 10px;
  border: 1px solid grey;
}

#randomize_div {
  margin: 10% 2% 0 auto;
}

#random_btn {
  color: white;
  font-size: 1.25em;
  width: 200px;
  height: 100%;
  border-radius: 15px;
  background: linear-gradient(217deg, rgba(255, 0, 0, .8), rgba(255, 0, 0, 0) 70.71%),
  linear-gradient(127deg, rgba(0, 255, 0, .8), rgba(0, 255, 0, 0) 70.71%),
  linear-gradient(336deg, rgba(0, 0, 255, .8), rgba(0, 0, 255, 0) 70.71%);
}

#random_btn:hover {
  background-color: grey;
  color: black;
  transform: scale(1.05);
}

.input_palette {
  border: 1px solid black;
  height: 40px;
  border-radius: 15px 0 0 15px;
}

.input_palette-btn {
  border: 1px solid black;
  width: 35px;
  height: 43px;
  border-radius: 0 15px 15px 0;
}

.input_palette-btn:hover {
  background-color: darkolivegreen;
  color: white;
  transform: scale(1.1);
}