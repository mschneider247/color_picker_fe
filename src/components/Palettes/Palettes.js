import React, { Component } from 'react';
import './Palettes.css';

class Palettes extends Component {
  constructor() {
    super();
    this.state = {
      currentPalette: {}
    }
  }

  render () {
    return(
      <section>
        <div id="add_new_palette">
          <div>
            <h2 id="palettes">Palettes:</h2>
            <h3>Add A New Palette:</h3>
            <input className="input_palette" placeholder="    Palette Name" onChange={this.props.updatePaletteName} type='text'/>
            <button className="input_palette-btn" onClick={this.props.postPalette}>+</button>
            <br />
          </div>
          <div id="randomize_div">
            <button id="random_btn" onClick={this.props.randomizeColors}>Randomize Palette</button>
          </div>
        </div>
        <div id="palettes_div">
          {this.props.palettes}
        </div>
      </section>
    )
  }
};

export default Palettes;