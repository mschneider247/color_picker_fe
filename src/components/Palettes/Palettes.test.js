import React from 'react';
import { shallow } from 'enzyme';
import Palettes from './Palettes';


describe('Palettes component', () => {
	let wrapper;
	let mockPostPalette = jest.fn();
	let mockUpdatePaletteName = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Palettes updatePaletteName={mockUpdatePaletteName} postPalette={mockPostPalette} />)
  });

  it('App should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state and invoke updatePaletteName when updateState is called', async () => {
  	let mockEvent = { target: { value: 'Name' } };
  	wrapper.instance().updateState(mockEvent);
  	await expect(wrapper.state('paletteName')).toEqual('Name');
  	expect(mockUpdatePaletteName).toHaveBeenCalledWith('Name');
  });

  it('should update state and invoke postPalette when addNewPalette is called', async () => {
  	let mockEvent = { target: { value: 'Name' } };
  	wrapper.instance().addNewPalette();
  	await expect(wrapper.state('paletteName')).toEqual('');
  	expect(mockUpdatePaletteName).toHaveBeenCalled();
  });
});