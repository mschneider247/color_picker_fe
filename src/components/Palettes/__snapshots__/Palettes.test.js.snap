// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palettes component Palettes should match its snapshot 1`] = `
<section>
  <div
    id="add_new_palette"
  >
    <div>
      <h2
        id="palettes"
      >
        Palettes:
      </h2>
      <h3>
        Add A New Palette:
      </h3>
      <input
        className="input_palette"
        onChange={[Function]}
        placeholder="    Palette Name"
        type="text"
        value=""
      />
      <button
        className="input_palette-btn"
        onClick={[Function]}
      >
        +
      </button>
      <br />
    </div>
    <div
      id="randomize_div"
    >
      <button
        id="random_btn"
      >
        Randomize Palette
      </button>
    </div>
  </div>
  <div
    id="palettes_div"
  />
</section>
`;
